src/hotspot/cpu/riscv/assembler_riscv.hpp

```diff
@@ -46,8 +46,9 @@
 class Argument {
  public:
   enum {
-    n_int_register_parameters_c   = 8, // x10, x11, ... x17 (c_rarg0, c_rarg1, ...)
-    n_float_register_parameters_c = 8, // f10, f11, ... f17 (c_farg0, c_farg1, ... )
+    n_int_register_parameters_c   = 8,   // x10, x11, ... x17 (c_rarg0, c_rarg1, ...)
+    n_float_register_parameters_c = 8,   // f10, f11, ... f17 (c_farg0, c_farg1, ... )
+    n_vector_register_parameters_c = 8,  // v8, v9, ... v15
```

Comment:
I know vector registers are not used for passing arguments or return values by the RISCV ABI for now. But I guess it might be better and consistent to align with the numbering of integer and floating-point argument registers? That is v10 - v17.

Reply:
Ah, I think I missed that. I was reading psABI spec 1.0 release. Thanks for this info.

Reply:
Thanks Robbin for helping explaining!

minor correction: v1-v7/v24-v31 = callee saved

Reply:
Yes, sorry what I meant, updated.

Reply:
Then why would we put a constraint on the number of argument vector registers here (v8-v15 instead of v8-v23)? Could we just support all of them, i.e., v8-v23 to comply with the RISC-V psABI?

Reply:
There is no strong reason, just it's sufficient for current implementation.

Maybe it's better to use them all in case in the future some other code touch the limit unnecessarily. I'll change it to use all arg v regs.

Reply:
Thanks for the update. Glad to see it's now fully compliant with the new calling convention variant of RISC-V psABI. I think the code will be easier for others to understand at the same time.